src/logic/events/AccessibilityEvents.swift

```diff
@@ -130,9 +136,9 @@ fileprivate func focusedWindowChanged(_ element: AXUIElement, _ pid: pid_t) thro
 
 fileprivate func windowDestroyed(_ element: AXUIElement) throws {
     let wid = try element.cgWindowId()
-    DispatchQueue.main.async {
-        if let index = (Windows.list.firstIndex { $0.isEqualRobust(element, wid) }) {
-            let window = Windows.list[index]
+    if let index = (Windows.list.firstIndex { $0.isEqualRobust(element, wid) }) {
+        let window = Windows.list[index]
+        DispatchQueue.main.async {
```

Comment:
I think it's dangerous to call objects like `Windows.list` from the AX background thread. That's why all the functions in this file have a pattern of doing a few lines of AX calls (all the `try`s), then doing work inside a `DispatchQueue.main` block.

Reply:
Fixed.